gufe/settings/models.py

```diff
@@ -0,0 +1,143 @@
+    """
+    See the `SMIRNOFF Electrostatics specification <https://openforcefield.github.io/standards/standards/smirnoff/#electrostatics>`_
+    for more details.
+    """
+    scale12: float = 0.0
+    scale13: float = 0.0
+    scale14: float = 0.833333
+    scale15: float = 1.0
+
+
+class VdWSettings(SettingsBaseModel):
+    """Settings for van der Waals force.
+
+    See the `SMIRNOFF vdW specification <https://openforcefield.github.io/standards/standards/smirnoff/#vdw>`_
+    for more details.
+    """
+
+    combining_rules: Literal["Lorentz-Berthelot"]
+    potential: Literal["Lennard-Jones-12-6"]
+    scale: vdWScale
+    long_range_dispersion: Literal["isotropic"]
+    cutoff: FloatQuantity["angstrom"] = Field(9.0, description="Default units are angstroms")
+    switch_width: FloatQuantity["angstrom"] = Field(1.0, description="Default units are angstroms")
+    method: Literal["cutoff"]
+
+
+class ElectrostaticSettings(SettingsBaseModel):
+    """Settings for electrostatics.
+
+    See the `SMIRNOFF Electrostatics specification <https://openforcefield.github.io/standards/standards/smirnoff/#electrostatics>`_
+    for more details.
+
+    """
+
+    # Tricky since this allows functions https://openforcefield.github.io/standards/standards/smirnoff/#electrostatics
+    periodic_potential: Union[Literal["Ewald3D-ConductingBoundary"], str]
+    nonperiodic_potential: Union[Literal["Coulomb"], str]
+    exception_potential: Union[Literal["Coulomb"], str]
+    scale: ElectrostaticScale
+    cutoff: FloatQuantity["angstrom"] = Field(None, description="Default units are in angstroms")
+    switch_width: FloatQuantity["angstrom"] = Field(None, description="Default units are in angstroms")
+    solvent_dielectric: Optional[float]
+
+
+class GBSASettings(SettingsBaseModel):
+    """Settings for Generalized-Born surface area (GBSA) implicit solvent parameters.
+
+    See the `SMIRNOFF GBSA specification <https://openforcefield.github.io/standards/standards/smirnoff/#gbsa>`_
+    for more details.
+    """
+
+    gb_model: str = "OBC1"
+    solvent_dielectric: float = 78.5
+    solute_dielectric: float = 1
+
+
+class ForcefieldSettings(SettingsBaseModel):
+    """Settings for the forcefield.
+    """
+
+    # Metadata
+    date: Optional[date]
+    author: Optional[str]
+
+    # These should also allow None
+    vdW: VdWSettings
+    electrostatics: ElectrostaticSettings
+    gbsa: GBSASettings
+
+
+class ThermoSettings(SettingsBaseModel):
+    """Settings for thermodynamic parameters.
+
+    No checking is done to ensure a valid thermodynamic ensemble is possible.
+    """
+
+    temperature: FloatQuantity["kelvin"] = Field(None, description="Simulation temperature, default units kelvin")
+    pressure: FloatQuantity["standard_atmosphere"] = Field(None, description="Simulation pressure, default units standard atmosphere (atm)")
+    volume: FloatQuantity["nm**3"] = Field(None, description="Simulation volume, default units are cubic nanometres")
```

Comment:
I'm not sure volume is right here.
1) Having volume as nm^3 is correct, but really you'd want the box vectors which are constant
2) The box vectors are more part of the chemical composition (ChemicalSystem half of things) rather than the Protocol definition.

Maybe constant V is actually denoted by Pressure not being set.

Reply:
Cool we can drop it and let the protocol handle the logic for that

Reply:
I'm not sure I agree here, this needs more discussion. 1) I don't think we can assume folks might not use these settings outside of gufe components, 2) it makes no sense to define these ensemble parameters without defining volume. You either go make the call that they aren't ensemble parameters and have incomplete ones, or you have everything.

Reply:
Maybe it's a little too much thermodynamics, but I like the idea that `pressure=None` implies constant volume, `temperature=None` implies constant energy, etc. Value of volume is determined by the unit cell, no? And usually that's what users have actually set -- do we require them to calculate volume and set it here as well?

I think the biggest risk is that someone might accidentally forget to set pressure or such when they meant to, leading to using the wrong ensemble. I'm not sure if we want to catch that, but the easy way to do it would be to make the default value be a custom sentinel object, so you have to explicitly set `None` if that's what you mean.

Reply:
They may or they may not set it. But you could envision someone using these settings as part of a workflow. It depends on if you want these settings to be settings that can be independent (in which case you need to cover all thermodynamic parameters) or that has to exist within a predefined conception of how the settings will be applied (in which case you can skimp out on certain things).

Reply:
What if for now we just kept `temperature` and `pressure`, keep the default as `None` and see how things go after an iteration on using this in a protocol?